src/deploy/index.ts

```diff
@@ -77,6 +78,16 @@ export const deploy = async function (
     }
   }
 
+  if (
```

Comment:
we need to remove the functions unshift in frameworks/index#prepareFrameworks for this logic to work

Reply:
Oh, really good point. I think we should just remove the guard against `!targetName.includes("functions")` to simplify things.

Reply:
removing the guard would break `--only hosting,functions` as it would then append to only

Reply:
Actually, we need to remove the guard. 

`--only hosting,functions` will still be `--only hosting,functions` after ensureTargeted, but `--only hosting,functions:someOtherFunction` may need to be `--only hosting,functions:ssr,functions:someOtherFunction`